salt/modules/junos.py

```diff
@@ -380,15 +396,24 @@ def commit(**kwargs):
             else:
                 ret['message'] = 'Commit failed.'
                 ret['out'] = False
+                conn.cu.rollback()
         except Exception as exception:
             ret['out'] = False
             ret['message'] = \
                 'Commit check succeeded but actual commit failed with "{0}"' \
                 .format(exception)
+            conn.cu.rollback()
     else:
         ret['out'] = False
         ret['message'] = 'Pre-commit check failed.'
         conn.cu.rollback()
+
+    if op['unlock']:
+        try:
+            conn.cu.unlock()
+        except jnpr.junos.exception.UnlockError:
+            pass
```

Comment:
This is pep20 violation which even I am not in favour of 
https://www.python.org/dev/peps/pep-0020/

"Errors should never pass silently."